src/lib/import_export/csv_parser.cpp

```diff
@@ -136,7 +136,16 @@ bool CsvParser::_find_fields_in_chunk(std::string_view csv_content, const Table&
     from = pos + 1;
     const char elem = csv_content.at(pos);
 
-    if (elem == _csv_config.quote) in_quotes = !in_quotes;
+    // Make sure to "toggle" in_quotes ONLY if the quotes are not part of the string (i.e. escaped)
+    if (elem == _csv_config.quote) {
+      bool escaped = false;
+      if (elem != _csv_config.escape) {
```

Comment:
How does this check make sense if two lines above, we already found `elem` to be `_csv_config.quote`? Is this only about escaping with double quotes? If so, it needs a comment.

Reply:
Basically it covers the case where the quote char is equal to the escape char (or rather where that's not the case). I'll update the clause to
`if (_csv_config.quote != _csv_config.escape)`
that should make that more clear.